lib/caching-precompiler.js

```diff
@@ -30,15 +31,26 @@ CachingPrecompiler.prototype._factory = function (cacheDir) {
 		// Extract existing source maps from the code.
 		var sourceMap = convertSourceMap.fromSource(code) || convertSourceMap.fromMapFileSource(code, path.dirname(filename));
 
-		return {
-			presets: [presetStage2, presetES2015],
-			plugins: [powerAssert, transformRuntime],
+		var baseOptions = {
 			filename: filename,
 			sourceMaps: true,
 			ast: false,
-			babelrc: false,
-			inputSourceMap: sourceMap && sourceMap.toObject()
+			inputSourceMap: sourceMap && sourceMap.toObject(),
+			babelrc: false
 		};
+
+		if (!babelConfig || babelConfig === 'default') {
+			return objectAssign({}, baseOptions, {
```

Comment:
No need for the empty object here, `baseOptions` is already fresh so it's safe to mix other properties into it.